presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -478,28 +485,69 @@ private void testRoundTripType(Type type, List<?> readValues)
 
     public void testRoundTripTypes(List<Type> types, List<List<?>> readValues, List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters)
             throws Exception
+    {
+        testRoundTripTypes(types, readValues, filters, ImmutableList.of());
+    }
+
+    public void testRoundTripTypes(List<Type> types, List<List<?>> readValues, List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters, List<List<Integer>> expectedFilterOrder)
+            throws Exception
     {
         assertEquals(types.size(), readValues.size());
+        if (expectedFilterOrder.size() > 0) {
+            assertEquals(filters.size(), expectedFilterOrder.size());
+        }
 
         // forward order
-        assertRoundTrip(types, readValues, filters);
+        assertRoundTrip(types, readValues, filters, expectedFilterOrder);
 
         // reverse order
         if (reverseTestsEnabled) {
-            assertRoundTrip(types, Lists.transform(readValues, OrcTester::reverse), filters);
+            assertRoundTrip(types, Lists.transform(readValues, OrcTester::reverse), filters, ImmutableList.of());
```

Comment:
I don't think this would work if we did that. We'd have to apply the reordering logic in testRoundTripTypesWithOrder. And still, I'm not sure it'll work. (The flag is currently false, presumably because other things break when turning it on?)